### PR TITLE
add medication modal fixes

### LINKED ISSUE #130: [FE] More add medication redesigning
### Description

figma links for stuff that has to be changed: 

https://www.figma.com/design/D9bT2htLRRJPX6A8tNlOBU/iCAN-Fall-25?node-id=14993-9279&t=sTeaYQdi6vvhP4AX-4 (the repeat on is centered and it shouldnt be)
https://www.figma.com/design/D9bT2htLRRJPX6A8tNlOBU/iCAN-Fall-25?node-id=14993-9383&t=sTeaYQdi6vvhP4AX-4 (the checkbox)
https://www.figma.com/design/D9bT2htLRRJPX6A8tNlOBU/iCAN-Fall-25?node-id=15001-676&t=sTeaYQdi6vvhP4AX-4 (error message)
https://www.figma.com/design/D9bT2htLRRJPX6A8tNlOBU/iCAN-Fall-25?node-id=15012-825&t=sTeaYQdi6vvhP4AX-4 (the medication tile designing)
https://www.figma.com/design/D9bT2htLRRJPX6A8tNlOBU/iCAN-Fall-25?node-id=15143-9285&t=sTeaYQdi6vvhP4AX-4 (the add medication button when there are no created medicine yet)

### Acceptance Criteria

- [ ] Design signoff
- [ ] preserved functionality

### Other Notes

- Pointers, tips, and links to pieces of code or resources that might help in solving this ticket

### PR DESCRIPTION
**Pull request by KashGiannis34**
closes #130

## More add medication modal changes

Issue Number(s): #130 

What does this PR change and why?
Small visual changes to add medication modal. Updated medication card on medication page to better depict next dosage time. Changed Add New button for empty medication page.

### Checklist

- [ ] Requirements have been implemented
- [ ] Acceptance criteria is met
- [ ] Database schema docs have been updated or are not necessary
- [ ] Code follows design and style guidelines
- [ ] Commits follow guidelines (concise, squashed, etc)
- [ ] Relevant reviewers (Senior Dev/EM/Designers) have been assigned to this PR

### Related PRs

- #130

### Testing

Test out what different dosage times cause the "Next dosage" time to appear as in the medication card. there may be a lot of edge cases for that, esp for the month option.

### CODE REVIEW

File: src/components/ui/MedicationCard.tsx [hunk 1]
```diff
@@ -17,43 +17,186 @@ export default function MedicationCard({
...
+        "Sunday",
+        "Monday",
+        "Tuesday",
+        "Wednesday",
+        "Thursday",
+        "Friday",
+        "Saturday",
+      ];
+      return days[dayIndex];
+    };
```
[the remark this answers is not in the document]
Comment (rgodha24) [reply src/components/ui/MedicationCard.tsx]: ok good point

**Review by rgodha24 — APPROVED** [review]
